linode/manager
# change: [M3-7232] Rename isPropValid and update codebase & documentation

## Description 📝
In our styled components, we have the ability to prevent the forwarding of custom props that could end up being passed as an invalid html attribute in the dom. To that effect we have a utility to use in `shouldForwardProp` that was named `isPropValid` but this name caused a bit of confusion since it made some assumption about the validity of the prop itself, while its intent is to only build an array of props to exclude from forwarding regardless of their semantic validity.

Note: the `omittedProps` name has been discussed and vetted with @jaalah-akamai 

## Changes  🔄
- Rename `isPropValid` to `omittedProps`
- Add unit test to the utility
- Update imports and instances with new name through the codebase
- Update documentation

## Preview 📷
There is no visual change brought by this PR, nor should there be any functionality change

## How to test 🧪

### Prerequisites
- Pull and run code locally 

### Verification steps 
- Verify there's no regression with the forwarding of our props
- Confirm documentation reflects our new standards and the usage of `omittedProps`

## As an Author I have considered 🤔

*Check all that apply*


---
## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**
- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---
